### PR TITLE
ci: lint and build workflows

### LINKED ISSUE #4: ci workflows

### PR DESCRIPTION
**Pull request by jasonappah**
This PR adds an ESLint and Prettier config that should work with TypeScript, and adds workflows for building and linting on every push and PR. Closes #4.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: actions/checkout@v2
+        name: Clone repo
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '14'
+        name: Set up Node
+      - run: npm ci 
+        name: Install dependencies
+      - run: npm run build
+        name: Build
```
Comment (zaida04) [inline .github/workflows/build.yml]:
```suggestion
        name: Build
        
```
EOF

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          echo "::set-output name=all::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | xargs)"
+          echo "::set-output name=ts::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep .ts$ | xargs)"
+      - if: ${{ steps.changes.outputs.all }}
+        name: Commit changes
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "GitHub Actions"
+          git add .
+          git commit -m "style: lint and format"
+          git push
```
Comment (zaida04) [inline .github/workflows/lint.yml]: We don't want to have to fight the linter workflow in the event that ESLint/Prettier is enforcing a change that we do not want. I'd rather we not have the workflow automatically commit fixes.

Also please add an EOF to the end of this file

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+
+    "printWidth": 90,
+    "tabWidth": 4,
+    "useTabs": false,
+    "semi": false,
```
Comment (zaida04) [inline .prettierrc]:
```suggestion
    "semi": true,
```
This is to avoid ASI failures: https://prettier.io/docs/en/rationale.html#semicolons.

**Review by zaida04 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
```
Comment (zaida04) [inline .github/workflows/build.yml]:
```suggestion
      - name: Clone repo
        uses: actions/checkout@v2
      - name: Set up Node
        uses: actions/setup-node@v1
        with:
          node-version: '14'
      - name: Install dependencies
        run: npm ci 
      - name: Build
        run: npm run build
```

Name should come first so people know off the bat what each step does.

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      - uses: actions/checkout@v2
+        name: Clone repo
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '14'
+        name: Set up Node
+      - run: npm ci
+        name: Install dependencies
+      - run: npm run lint
+        name: Run Linter
```
Comment (zaida04) [inline .github/workflows/lint.yml]:
```suggestion
      - name: Clone repo
        uses: actions/checkout@v2
      - name: Set up Node
        uses: actions/setup-node@v1
        with:
          node-version: '14'
      - name: Install dependencies
        run: npm ci
      - name: Run Linter
        run: npm run lint
```

**Review by zaida04 — CHANGES_REQUESTED** [review]
After this review is resolved, I will approve this PR.

**Review by zaida04 — APPROVED** [review]

File: components/head.js [hunk 6]
```diff
@@ -16,7 +17,8 @@ export default function HeadObject({children}) {
             <meta name="description" content={description} />
             <meta name="keywords" content={keywords} />
             <meta name="author" content={author} />
-            <meta property="og:url" content="put it in this quotes" /> {/* This is where you put the domain */}
+            <meta property="og:url" content="put it in this quotes" />{" "}
```
Comment (sarthaktexas) [inline components/head.js]: lmao can you remove this {" "} tag

**Comment by sarthaktexas** [comment]
I don't totally love the fact of Typescript, can you still use jsx just like normal for beginners?

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
answer the above question and fix my very minuscule comment

**Comment by zaida04** [comment]
> I don't totally love the fact of Typescript, can you still use jsx just like normal for beginners?

Looking through the `changed files`, all of the workflows, eslint, and tsconfig files have full support for js and jsx. It just adds support for ts and tsx files as well.

**Review by zaida04 — DISMISSED** [review]

**Comment by jasonappah** [comment]
Yep, it should allow both regular jsx and tsx.

**Comment by sarthaktexas** [comment]
Great, I'm completely fine with this. I was gonna add prettier but thanks for doing that!

File: components/head.js [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
-    const title = "The Perfect Next.js Theme";
-    const description = "This is the perfect Next.js theme that you should use instead of create-next-app.";
-    const keywords = "put some fun keywords here";
-    const author = "Sarthak Mohanty";
-    const twitter = "@sarthaktexas";
-    //const image = "/ogimage.png"; // This is your OpenGraph image
+export default function HeadObject({ children }) {
+    const title = "The Perfect Next.js Theme"
+    const description =
+        "This is the perfect Next.js theme that you should use instead of create-next-app."
```
Comment (sarthaktexas) [inline components/head.js]: why is this on a new line?!?

**Comment by sarthaktexas** [comment]
Apart from that I think everything's fine, I'll approve it

File: components/head.js (hunk 7 above)
Comment (zaida04) [reply components/head.js]: https://github.com/ThinkFiveable/open/blob/8fb8e0ba9c4750bf91c7e364d7dcb9ad137f0d03/.prettierrc#L3

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
Everything is good, but there's merge conflicts, fix them, and I'll approve

**Comment by sarthaktexas** [comment]
did i forget to do an angular commit message 😭 @zaida04 can we still merge this :) i promise it won't happen again and i don't know how to rebase this cause it's not mine

**Comment by zaida04** [comment]
Please resolve these merge conflicts.

**Comment by jasonappah** [comment]
Alright, I think all the conflicts are fixed.

**Review by zaida04 — DISMISSED** [review]

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
I am _so_ sorry for doing this again. Looks like you added format commands, is this automatic or do you have to do it manually. If the latter, can you put it in the README?

**Comment by jasonappah** [comment]
The linter doesn't run automatically, I can add the commands to the readme.

**Review by zaida04 — DISMISSED** [review]

**Review by zaida04 — APPROVED** [review]

**Review by sarthaktexas — APPROVED** [review]

**Pull request merged**